### PR TITLE
macOS: Add tabs for Homebrew/MacPorts and for 3.13+/3.11-3.12/3.8-3.10

### PR DESCRIPTION
**Pull request by hugovk**
For example:

image

----
📚 Documentation preview 📚: https://cpython-devguide--1317.org.readthedocs.build/getting-started/setup-building/#install-dependencies

### CODE REVIEW

File: getting-started/setup-building.rst [hunk 1]
```diff
@@ -690,54 +690,64 @@ on Linux, macOS and iOS.
    manager, like Homebrew_ or MacPorts_, and then add the appropriate paths
    for the header and library files to your ``configure`` command.
 
-   For example, with **Homebrew**, install the dependencies::
+   .. tab:: Homebrew
 
-      $ brew install pkg-config openssl@3.0 xz gdbm tcl-tk mpdecimal
+      For example, with **Homebrew**, install the dependencies::
```
Comment (erlend-aasland) [inline getting-started/setup-building.rst:695]:
```suggestion
      For **Homebrew**, install dependencies using ``brew``::
```

File: getting-started/setup-building.rst [hunk 2]
```diff
@@ -690,54 +690,64 @@ on Linux, macOS and iOS.
...
 
-      $ GDBM_CFLAGS="-I$(brew --prefix gdbm)/include" \
-         GDBM_LIBS="-L$(brew --prefix gdbm)/lib -lgdbm" \
-         ./configure --with-pydebug \
-                     --with-system-libmpdec \
-                     --with-openssl="$(brew --prefix openssl@3.0)"
+      .. tab:: Python 3.13+
 
-   For Python 3.11 and 3.12::
+         Then, for Python 3.13 and newer, run ``configure``::
```
Comment (erlend-aasland) [inline getting-started/setup-building.rst:701]:
```suggestion
         For Python 3.13 and newer::
```

File: getting-started/setup-building.rst [hunk 3]
```diff
@@ -690,54 +690,64 @@ on Linux, macOS and iOS.
...
-   And finally, run ``make``::
+            $ GDBM_CFLAGS="-I$(brew --prefix gdbm)/include" \
+               GDBM_LIBS="-L$(brew --prefix gdbm)/lib -lgdbm" \
+               ./configure --with-pydebug \
+                        --with-openssl="$(brew --prefix openssl@3.0)"
 
-      $ make -s -j2
+      .. tab:: Python 3.8-3.10
+
+         Or, for Python 3.8 through 3.10::
```
Comment (erlend-aasland) [inline getting-started/setup-building.rst:720]:
```suggestion
         For Python 3.8 through 3.10::
```

File: getting-started/setup-building.rst [hunk 4]
```diff
@@ -690,54 +690,64 @@ on Linux, macOS and iOS.
...
+            $ CPPFLAGS="-I$(brew --prefix gdbm)/include -I$(brew --prefix xz)/include" \
+               LDFLAGS="-L$(brew --prefix gdbm)/lib -L$(brew --prefix xz)/lib" \
+               ./configure --with-pydebug \
+                           --with-openssl="$(brew --prefix openssl@3.0)" \
+                           --with-tcltk-libs="$(pkg-config --libs tcl tk)" \
+                           --with-tcltk-includes="$(pkg-config --cflags tcl tk)"
+
+   .. tab:: MacPorts
+
+      Alternatively, with **MacPorts**::
```
Comment (erlend-aasland) [inline getting-started/setup-building.rst:731]:
```suggestion
      For **MacPorts**, install dependencies using ``port``::
```

File: getting-started/setup-building.rst [hunk 5]
```diff
@@ -690,54 +690,64 @@ on Linux, macOS and iOS.
...
+
+         $ sudo port install pkgconfig openssl xz gdbm tcl tk +quartz mpdecimal
 
-   Alternatively, with **MacPorts**::
+      .. tab:: Python 3.13+
 
-      $ sudo port install pkgconfig openssl xz gdbm tcl tk +quartz mpdecimal
+         Then, for Python 3.13 and newer, run ``configure``::
```
Comment (erlend-aasland) [inline getting-started/setup-building.rst:737]:
```suggestion
         For Python 3.13 and newer::
```

File: getting-started/setup-building.rst [hunk 6]
```diff
@@ -690,54 +690,64 @@ on Linux, macOS and iOS.
...
+                           --with-system-libmpdec
 
-      $ GDBM_CFLAGS="-I$(dirname $(dirname $(which port)))/include" \
-         GDBM_LIBS="-L$(dirname $(dirname $(which port)))/lib -lgdbm" \
-         ./configure --with-pydebug \
-                     --with-system-libmpdec
+      .. tab:: Python 3.11-3.12
 
-   Or, for Python 3.11 and 3.12, run ``configure``::
+         Or, Python 3.11 and 3.12, run ``configure``::
```
Comment (erlend-aasland) [inline getting-started/setup-building.rst:746]:
```suggestion
         For Python 3.11 and 3.12::
```

File: getting-started/setup-building.rst [hunk 7]
```diff
@@ -690,54 +690,64 @@ on Linux, macOS and iOS.
...
 
-      $ GDBM_CFLAGS="-I$(brew --prefix gdbm)/include" \
-         GDBM_LIBS="-L$(brew --prefix gdbm)/lib -lgdbm" \
-         ./configure --with-pydebug \
-                     --with-openssl="$(brew --prefix openssl@3.0)"
+            $ GDBM_CFLAGS="-I$(brew --prefix gdbm)/include" \
+               GDBM_LIBS="-L$(brew --prefix gdbm)/lib -lgdbm" \
+               ./configure --with-pydebug \
+                           --with-system-libmpdec \
+                        --with-openssl="$(brew --prefix openssl@3.0)"
```
Comment (ezio-melotti) [inline getting-started/setup-building.rst:707]:
```suggestion
            $ GDBM_CFLAGS="-I$(brew --prefix gdbm)/include" \
               GDBM_LIBS="-L$(brew --prefix gdbm)/lib -lgdbm" \
               ./configure --with-pydebug \
                           --with-system-libmpdec \
                           --with-openssl="$(brew --prefix openssl@3.0)"
```

File: getting-started/setup-building.rst [hunk 8]
```diff
@@ -690,54 +690,64 @@ on Linux, macOS and iOS.
...
-                     --with-openssl="$(brew --prefix openssl@3.0)" \
-                     --with-tcltk-libs="$(pkg-config --libs tcl tk)" \
-                     --with-tcltk-includes="$(pkg-config --cflags tcl tk)"
+         For Python 3.11 and 3.12::
 
-   And finally, run ``make``::
+            $ GDBM_CFLAGS="-I$(brew --prefix gdbm)/include" \
+               GDBM_LIBS="-L$(brew --prefix gdbm)/lib -lgdbm" \
+               ./configure --with-pydebug \
+                        --with-openssl="$(brew --prefix openssl@3.0)"
```
Comment (ezio-melotti) [inline getting-started/setup-building.rst:716]:
```suggestion
            $ GDBM_CFLAGS="-I$(brew --prefix gdbm)/include" \
               GDBM_LIBS="-L$(brew --prefix gdbm)/lib -lgdbm" \
               ./configure --with-pydebug \
                           --with-openssl="$(brew --prefix openssl@3.0)"
```

File: getting-started/setup-building.rst [hunk 9]
```diff
@@ -690,54 +690,64 @@ on Linux, macOS and iOS.
...
 
-      $ make -s -j2
+      .. tab:: Python 3.8-3.10
+
+         For Python 3.8 through 3.10::
```
Comment (ezio-melotti) [inline getting-started/setup-building.rst:720]:
```suggestion
         For Python 3.8, 3.9, and 3.10::
```
I find this more explicit and understandable.

**Review by ezio-melotti — APPROVED** [review]

**Pull request merged**